Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/AssetBrowserModel.cpp

```diff
@@ -460,6 +493,31 @@ namespace AzToolsFramework
             index = createIndex(row, column, entry);
             return true;
         }
+
+        void AssetBrowserModel::WatchForExpectedAssets(AssetBrowserEntry* entry)
+        {
+            const AZStd::string& fullpath = AZ::IO::Path(entry->GetFullPath()).AsPosix();
+            if (m_newlyCreatedAssetPathsToCreatorBusIds.contains(fullpath))
+            {
+                if (m_newlyCreatedAssetPathsToCreatorBusIds[fullpath] != AZ::Crc32())
+                {
+                    m_assetEntriesToCreatorBusIds[entry] = m_newlyCreatedAssetPathsToCreatorBusIds[fullpath];
+                }
+
+                m_newlyCreatedAssetPathsToCreatorBusIds.erase(fullpath);
+
+                QTimer::singleShot(0, this,
+                    [&, entry]()
```

Comment:
nit: I'm not sure why you're capturing entry (a pointer) by value but everything else by reference. Is this to call attention to using entry directly/intentionally? If so, can you change the '&' capture to just be a 'this' capture for your emit, so that the reader doesn't need to check what you're using in the captured &?

Reply:
I totally agree that it seems out of place at first glance, but passing the copy is necessary here. If you pass the ptr by reference then it will be a reference to a ptr that was created on the stack and has fallen out of scope, so that's going to cause an access violation when you try to dereference it at the top of the next Qt event loop.

As far as capturing `this` by reference and then explicitly capturing other variables, that totally seems more clear here. I did some reading about lambda captures and think I know what might help. Let me know if you think it needs changes still after the next commit.